task-maker-format/src/ioi/sanity_checks/statement.rs

```diff
@@ -417,3 +487,61 @@ fn find_statement_pdf(task: &IOITask) -> Option<PathBuf> {
     }
     None
 }
+
+/// Search for a language-specific statement file, returning its path or None if it doesn't exists.
+///
+/// Will return the path even in case of broken links.
+fn find_language_statement_pdf(task: &IOITask, language: &str) -> Option<PathBuf> {
+    for path in &[
+        format!("statement/{language}.pdf"),
+        format!("testo/{language}.pdf"),
+    ] {
+        let path = task.path.join(path);
+        if path.exists() || path.read_link().is_ok() {
+            return Some(path);
+        }
+    }
+    None
+}
+
+/// Checks whether a file is a valid PDF file
+fn check_valid_pdf(path: &Path) -> Result<bool, Error> {
+    let mut file = std::fs::File::open(path)
+        .with_context(|| format!("Failed to open statement file at {}", path.display()))?;
+    let mut buf = [0u8; 4];
+
+    let valid = match file.read_exact(&mut buf) {
+        Err(_) => false,
+        Ok(_) => {
+            // check PDF magic number
+            &buf == b"%PDF"
+        }
+    };
+
+    Ok(valid)
+}
+
+/// Checks whether a file is known to git
+///
+/// If git is not present, there is no git repository, or no file is tracked at all
+/// this will behave as if the file is known.
+fn check_known_to_git(task: &IOITask, path: &Path) -> Result<bool, Error> {
```

Comment:
Might want to do `git ls-files <path>` and check for non-empty stdout...

Reply:
Isn't this the way it's already done?

Reply:
no, this seems to list all the files in the task dir, and then filter the output.

Btw, is it intended that if there's no files in the task dir, this returns true?

Reply:
It lists all files because this way it is possible to check if any files are being tracked at all. This was the original behaviour of the check as well.

I don't really know why it was intended to return true in directories without tracked files.
My best guess is that before adding the task file to git for the first time, it would make little sense to show the warning for untracked files.